Lib/test/test_zipfile/test_core.py

```diff
@@ -3447,6 +3447,100 @@ def test_too_short(self):
         self.assertEqual(
             b"zzz", zipfile._Extra.strip(b"zzz", (self.ZIP64_EXTRA,)))
 
+class StoreSeekReadTest(unittest.TestCase):
+    def test_store_seek_read(self):
+        class StatIO(io.RawIOBase):
```

Comment:
Don't we have a base class that you could use to avoid re-implementing all methods? (maybe in another file, like `test_memoryview` or `test_io`). Or maybe the Python implementation of `io` could already have a class ready for that.

Reply:
The purpose of this class is to count the total number of bytes read, which is used to check whether ZipExtFile will generate redundant reads after seek. I didn't find a class with similar functionality anywhere else.

Reply:
In `Lib/_pyio.py`, we may have something that already has the minimal interface. Buffers are implemented using pure Python, e.g., `_pyio.BytesIO()`. You can subclass it and tweak the methods you want to mock (possibly relying on using `super()` calls).

Reply:
Thanks. I will try.